app/controllers/groups_controller.rb

```diff
@@ -0,0 +1,85 @@
+class GroupsController < ApplicationController
+  before_action :authenticate_user!
+
+  # GET /groups or /groups.json
+  def index
+    @groups = current_user.groups
+  end
+
+  # GET /groups/1 or /groups/1.json
+  def show
+    @group = Group.find(params[:id])
+    @bills = Group.find(params[:id]).bills
+  end
+
+  # GET /groups/new
+  def new
+    # @groups = Group.all
+    @group = Group.new
+  end
+
+  def create
+    @group = current_user.groups.new(group_params)
+
+    if @group.save
+      redirect_to groups_path, notice: 'Group was successfully created.'
+    else
+      render :new
+    end
+  end
+
+  # # GET /groups/1/edit
+  # def edit
+  # end
+
+  # # POST /groups or /groups.json
+  # def create
+  #   @group = Group.new(group_params)
+
+  #   respond_to do |format|
+  #     if @group.save
+  #       format.html { redirect_to group_url(@group), notice: "Group was successfully created." }
+  #       format.json { render :show, status: :created, location: @group }
+  #     else
+  #       format.html { render :new, status: :unprocessable_entity }
+  #       format.json { render json: @group.errors, status: :unprocessable_entity }
+  #     end
+  #   end
+  # end
+
+  # # PATCH/PUT /groups/1 or /groups/1.json
+  # def update
```

Comment:
* Kindly remove all commented code as it makes your codebase look bulky and unprofessional